src/frontend/src/components/ForgotPassword/index.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { AppState } from '../../redux/store';
+import { logIn } from '../../redux/system/actions';
+import { SystemState } from '../../redux/system/types';
+import Logo from '../Logo';
+import { Link } from 'react-router-dom';
+
+interface Props {
+  system: SystemState;
+  logIn: typeof logIn;
+}
+interface State {
+  email: string;
+  emailHasInput: boolean;
+  emailIsValid: null | boolean;
+  missingInputs: null | boolean;
+}
+
+class ForgotPassword extends React.Component<Props, State> {
+  state: State = {
+    email: '',
+    emailHasInput: false,
+    emailIsValid: null, // email validation to be added
+    missingInputs: null
+  };
+
+  handleChange = (e: React.BaseSyntheticEvent) => {
+    // See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26635 for why we're
+    // using the "any" type.
+    this.setState<any>({
+      [e.target.id]: e.target.value
+    });
+  };
+
+  handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    this.validateForm();
+  };
+
+  emailCheck = (email: string) => {
+    return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
```

Comment:
Using a regex to validate an email address is notoriously fraught with difficulty, see https://stackoverflow.com/q/201323/1067145

I am not going to test your regex here. Just wanted to call this out and make sure the regex is relatively permissive and errs on the side of only rejecting emails that are definitely invalid.

Reply:
Hmm, this is only checking if the syntax is correct not if the address is actually "valid". It's fairly permissive but won't accept addresses with the format:
`_@_@._`

Reply:
Okay! Can you take the gist of what you wrote in your comment ^ and make that a comment in the code? It's hard for someone reading the code to make sense of a regex presented without any additional context.